lib/validation.js

```diff
@@ -76,32 +76,40 @@ function validateParam (validatorFunction, request, paramName) {
 
 function validate (context, request) {
   var params = validateParam(context[paramsSchema], request, 'params')
+  var schemaErrorFormatter = (context.schemaErrorFormatter || schemaErrorsText).bind(request)
+
   if (params) {
-    return wrapValidationError(params, 'params')
+    return wrapValidationError(params, 'params', schemaErrorFormatter)
   }
   var body = validateParam(context[bodySchema], request, 'body')
   if (body) {
-    return wrapValidationError(body, 'body')
+    return wrapValidationError(body, 'body', schemaErrorFormatter)
   }
   var query = validateParam(context[querystringSchema], request, 'query')
   if (query) {
-    return wrapValidationError(query, 'querystring')
+    return wrapValidationError(query, 'querystring', schemaErrorFormatter)
   }
   var headers = validateParam(context[headersSchema], request, 'headers')
   if (headers) {
-    return wrapValidationError(headers, 'headers')
+    return wrapValidationError(headers, 'headers', schemaErrorFormatter)
   }
   return null
 }
 
-function wrapValidationError (result, dataVar) {
+function wrapValidationError (result, dataVar, schemaErrorFormatter) {
   if (result instanceof Error) {
-    return result
+    return Object.assign(result, { statusCode: 400 })
+  }
+
+  try {
+    return Object.assign(schemaErrorFormatter(result, dataVar), {
+      validation: result,
+      validationContext: dataVar,
+      statusCode: 400
```

Comment:
The status code here is applyed here:
https://github.com/fastify/fastify/blob/7ac9da63f7d0ab03dd0548fdb52db03d3295f3e7/lib/handleRequest.js#L85